tiledb/sm/cpp_api/arrow_io_impl.h

```diff
@@ -639,6 +666,15 @@ void ArrowImporter::import_(
 
     query_->set_data_buffer(name, static_cast<void*>(p_data), data_num);
   }
+
+  if (typeinfo.nullable && arw_array->buffers[0] != nullptr) {
+    bitmap_to_bytemap(
+        const_cast<void*>(arw_array->buffers[0]), arw_array->length);
+    query_->set_validity_buffer(
+        name,
+        static_cast<uint8_t*>(const_cast<void*>(arw_array->buffers[0])),
+        arw_array->length);
```

Comment:
I would have `bitmap_to_bytemap` return a `vector<uint8_t>` and then pass that in rather than modifying the original buffers. Arrow structs are meant to be immutable, so I think we should keep that principle. 

```suggestion
    std::vector<uint8_t> casted_validity = bitmap_to_bytemap(
        const_cast<void*>(arw_array->buffers[0]), arw_array->length);
    query_->set_validity_buffer(
        name,
        casted_validity.data(),
        arw_array->length);
```

Reply:
I think I originally got these from George's work in libtiledbsoma, and the copy or variant I have in the R package has comments that the in-place mods are by design....

Reply:
After discussed with @ihnorton and @nguyenv
We own the buffer at this point, so we're not referring to the original buffer. The concern with returning a copy was the memory overhead.

Reply:
After discussing with @ihnorton and @nguyenv:
We own the buffer at this point, so we're not referring to the original buffer. The concern with returning a copy was the memory overhead.